pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -474,3 +475,57 @@ func TestRemediate(t *testing.T) {
 		}
 	}
 }
+
+func TestIsMaster(t *testing.T) {
+	masterMachine := machine("master")
+	masterMachine.Labels["sigs.k8s.io/cluster-api-machine-role"] = "master"
+	masterMachine.Labels["sigs.k8s.io/cluster-api-machine-type"] = "master"
+	masterMachine.Status = capiv1alpha1.MachineStatus{
+		NodeRef: &corev1.ObjectReference{
+			Namespace: "",
+			Name:      "master",
+		},
+	}
+	masterNode := node("master", true)
+	masterNode.Annotations = map[string]string{
+		"machine": fmt.Sprintf("%s/%s", namespace, "master"),
+	}
+	masterNode.Labels["node-role.kubernetes.io/master"] = ""
+
+	workerMachine := machine("worker")
+	workerMachine.Labels["sigs.k8s.io/cluster-api-machine-role"] = "worker"
+	workerMachine.Labels["sigs.k8s.io/cluster-api-machine-type"] = "worker"
+
+	workerMachine.Status = capiv1alpha1.MachineStatus{
+		NodeRef: &corev1.ObjectReference{
+			Namespace: "",
+			Name:      "worker",
+		},
+	}
+	workerNode := node("worker", true)
+	workerNode.Annotations = map[string]string{
+		"machine": fmt.Sprintf("%s/%s", namespace, "worker"),
+	}
+	workerNode.Labels["node-role.kubernetes.io/worker"] = ""
+
+	testCases := []struct {
+		machine  *capiv1alpha1.Machine
+		expected bool
+	}{
+		{
+			machine:  masterMachine,
+			expected: true,
+		},
+		{
+			machine:  workerMachine,
+			expected: false,
```

Comment:
Can drop this; `false` is the "zero" value.

Reply:
thanks thanks